base/genericmemory.jl

```diff
@@ -227,7 +227,7 @@ Memory{T}(x::AbstractArray{S,1}) where {T,S} = copyto_axcheck!(Memory{T}(undef,
 
 function _iterate_array(A::Union{Memory, Array}, i::Int)
     @inline
-    (i - 1)%UInt < length(A)%UInt ? (A[i], i + 1) : nothing
+    _checkbounds_array(Bool, A, i) ? (A[i], i + 1) : nothing
```

Comment:
Should this be
```suggestion
    _checkbounds_array(Bool, A, i) ? (@inbounds A[i], i + 1) : nothing
```
like the `iterate(s::CodeUnits, i=1)` method (which would then be _identical_)?